### ![Callysto.ca Banner](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-top.jpg?raw=true)

<a href="https://hub.callysto.ca/jupyter/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2Fcallysto%2Fhackathon&branch=master&subPath=SustainabilityOnMars/AmalTrack/ChallengeTemplate/challenge-template.ipynb&depth=1" target="_parent"><img src="https://raw.githubusercontent.com/callysto/curriculum-notebooks/master/open-in-callysto-button.svg?sanitize=true" width="123" height="24" alt="Open in Callysto"/></a>

# *Sustaining Life on Mars: Data Science Challenge*.

You’re a data scientist on a team of newly-arrived humans. While you were on Earth, you figured out how you could make the planet habitable. From growing food to clothing needs, you need to start building the framework for sustaining life on the red planet. 

Use data to answer questions such as:

1. What food do we need to bring?
        e.g. trees, seeds, genetically-modified foods
        
2. How do we feed people there?
        Consider: supply, manage, distribute, connect

7. What are essential key resources? 
        e.g. Electricity, oxygen, water, fuel, brick, plastics, steel, food. 

4. How do we decide who will go?
        e.g. population proportions, demographics, health, qualifications, genetic diversity

5. What forms of entertainment would people need? 
        e.g. music, books, pets, lego

6. What machines do we need? 
        e.g. cars, ships, fighter jets, rockets, computers, mobile phones.  
        
#### Choose one or more of these questions to answer, or come up with your own. Check out the example notebooks, and complete the sections in this notebook to answer your chosen question or questions

### Section I: About Me

Double-click this cell and tell us:
1. My name: Thierry Laforge

2. My e-mail address: 21laforgth@gmail.com

3. Why I picked this challenge: I know quite a bit of python and how to make GUIs and games but I have never done any data analysis or anything of the sort before. I feel as if this will help me get into data analysis and further my Python knowledge.

4. The question I picked: What are essential key resources? What will humans need to survive on Mars and how long will they survive off water supplies from earth?

### Section II: The data I used

1. Water Use and Stress
2. https://ourworldindata.org/water-use-stress#freshwater-use-by-region
3. To sustain life anywhere, water is an essential resource. Without it, humans can only survive for around 3 days! The dataset I chose would allow me to get the average daily water consumption in the OECD which I can then calculate how long a human can survive on Mars with a provided amount of water from earth.

In [1]:
# Use this cell to import libraries
import pandas as pd
import matplotlib.pyplot as plt
import cufflinks as cf

In [2]:
# Use this cell to read the data - use the tutorials if you are not sure how to do this
# Double-click to edit this cell. 
csv = "./OECD_freshwater_usage.csv"
df = pd.read_csv(csv, low_memory = False)

### Section III: Data Analysis and Visualization

### 👨🏽‍💻 Provide your code to explore and analyse your data

In [3]:
# Finding random facts
print("Most water that was used: %s L" % ("{:e}".format(df["Freshwater_use"].max() * 1000)))
print("Least water that was used: %s L" % ("{:e}".format(df["Freshwater_use"].min() * 1000)))
print("Average water use (per year): %s L" % ("{:e}".format(df["Freshwater_use"].mean() * 1000)))
print("Year that used the most water: %s" % (int(df["Year"].where(df['Freshwater_use'] == df["Freshwater_use"].max()).dropna().values[0])))
print("Year that used the most water: %s" % (int(df["Year"].where(df['Freshwater_use'] == df["Freshwater_use"].min()).dropna().values[0])))

Most water that was used: 9.500768e+14 L
Least water that was used: 1.149999e+14 L
Average water use (per year): 4.950768e+14 L
Year that used the most water: 2007
Year that used the most water: 1902


In [4]:
# If we send a Canadian astronaut to Mars, we can calculate how long the astronaut will be able to survive
# with a certain amount of water brought from earth. (Not recycled)

available_water = 53070.31 # The new SpaceX rocket is said to be able to carry 117,000 lbs of cargo.
                            # Assuming that the astronauts bring only water, we can convert 117,000 lbs to L.
can_avg = 329 # Litres
print("Days with water (Regular use): %s" % (available_water // can_avg))


Days with water (Regular use): 161.0


In [5]:
# So a human can survive 161 days on Mars using the water brought along from earth. But a lot of it is used on
# cooking and bathing. If we take those out of the equation, we can stretch this number.

can_avg_drink = 329 * 0.1 # Only 10% or so is actually used by drinking water and cooking.
print("Days with water (Just cooking and drinking): %s" % (available_water // can_avg_drink))

Days with water (Just cooking and drinking): 1613.0


In [6]:
# If a human can survive 1613 days by only drinking water and cooking, and the maximum amount of days one can
# go without water is 3 days, we can somehow stretch this number even thinner. If the astronaut only eats and
# drinks every other day, the new time would be...
can_avg_bare = 329 * 0.1 / 2
print("Days with water (Cooking and drinking every other day): %s" % (available_water // can_avg_bare))

Days with water (Cooking and drinking every other day): 3226.0


### Observations
Double-click this cell and write down at least 2 - 3 things or more you observed in your data through your analysis and visualizations.

1. The year that used up the most water was 2007 which was (9.500768e+14 L)
2. The year that used up the least amount of water was 1902 which was (1.149999e+14 L)
3. On average, all of the countries managed to use up around 4.950768e+14 L of water a year.
4. If one Canadian were to go to Mars on a SpaceX rocket with a cargo full of water, he/she would be able to
   last around 161 days living like how they did at home, 1613 days if they only drank and cooked, and 3226 days if    they only drank and cooked every other day.

### Section IV: Conclusion

It is crucial that you connect what you learned via the dataset to the main question(s) you are asking. 

Use this space to propose a solution to the question you picked. Make sure it is clear in your answer what area of development you chose to focus on and your proposed solution based on the dataset(s) you worked on. 

See our example notebooks for some inspiration on questions and solutions that you can develop using data.

Provide your analysis-driven result(s) here. Write down 2 - 3 things you learned from the data and how they help answer your main question(s). Also write down 2 - 3 things you learned from participating in this hackathon. 
Double-click to edit this cell. 

1. As a collective group in the OECD, on average we use around 4.95e+14 litres of water a day.
2. A Canadian astronaut would be able to survive for 3226 days at most assuming all goes accoding to plan while   ignoring some factors. (i.e sweating, excersize, evaporation, etc.)
1. Lots of data science and data analysis.
2. New libraries in python that I will be able to use in the future.

### [![Callysto.ca License](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-bottom.jpg?raw=true)](https://github.com/callysto/curriculum-notebooks/blob/master/LICENSE.md)